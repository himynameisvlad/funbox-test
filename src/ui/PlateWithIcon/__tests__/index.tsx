import { shallow } from 'enzyme';
import * as React from 'react';
import renderer from 'react-test-renderer';

import { PlateWithIcon } from '../index';

describe('PlateWithIcon - ui component', () => {
  let wrapper;
  const title = 'Test title';
  const children = <div />;
  const PlateWithIconComponent = <PlateWithIcon title={title}>{children}</PlateWithIcon>;

  beforeEach(() => {
    wrapper = shallow(PlateWithIconComponent);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(PlateWithIconComponent)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders children', () => {
    expect(wrapper.contains(children)).toEqual(true);
  });

  it('renders title', () => {
    expect(wrapper.find('span').text()).toEqual(title);
  });
})
