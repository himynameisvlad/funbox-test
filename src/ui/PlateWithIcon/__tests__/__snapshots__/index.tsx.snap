// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlateWithIcon - ui component renders correctly 1`] = `
<div
  className="fb-plate"
>
  <span>
    Test title
  </span>
  <div
    className="fb-plate__icon"
  >
    <div />
  </div>
</div>
`;
